extensions/core/interfaces/code/interface.vue

```diff
@@ -0,0 +1,161 @@
+
+// theme css
+// import 'codemirror/theme/monokai.css'
+
+// require active-line.js
+import'codemirror/addon/selection/active-line.js'
+
+// styleSelectedText - (styleSelectedText and css class .CodeMirror-selectedtext)
+import'codemirror/addon/selection/mark-selection.js'
+import'codemirror/addon/search/searchcursor.js'
+
+// hint - (extraKeys and hintOptions)
+import'codemirror/addon/hint/show-hint.js'
+import'codemirror/addon/hint/show-hint.css'
+import'codemirror/addon/hint/javascript-hint.js'
+import'codemirror/addon/selection/active-line.js'
+
+// highlightSelectionMatches - (highlightSelectionMatches)
+import'codemirror/addon/scroll/annotatescrollbar.js'
+import'codemirror/addon/search/matchesonscrollbar.js'
+import'codemirror/addon/search/searchcursor.js'
+import'codemirror/addon/search/match-highlighter.js'
+
+// keyMap - (matchBrackets)
+import'codemirror/mode/clike/clike.js'
+import'codemirror/addon/edit/matchbrackets.js'
+import'codemirror/addon/comment/comment.js'
+import'codemirror/addon/dialog/dialog.js'
+import'codemirror/addon/dialog/dialog.css'
+import'codemirror/addon/search/searchcursor.js'
+import'codemirror/addon/search/search.js'
+import'codemirror/keymap/sublime.js'
+
+// foldGutter - (foldGutter and gutters)
+import'codemirror/addon/fold/foldgutter.css'
+import'codemirror/addon/fold/brace-fold.js'
+import'codemirror/addon/fold/comment-fold.js'
+import'codemirror/addon/fold/foldcode.js'
+import'codemirror/addon/fold/foldgutter.js'
+import'codemirror/addon/fold/indent-fold.js'
+import'codemirror/addon/fold/markdown-fold.js'
+import'codemirror/addon/fold/xml-fold.js'
+
+import './code.css';
+import mixin from '../../../mixins/interface';
+
+export default {
+  name: "interface-code",
+  mixins: [mixin],
+  data () {
+    return {
+      code: "",   
+    }
+  },
+  watch: {
+    options: function () {
+      this.codemirror.setSize("100%", this.options.height);
+    }
+  },
+  computed: {
+    cmOptions() {
+      return {
+        tabSize: 4,
+        indentUnit: 4,
+        styleActiveLine: true,
+        lineNumbers: this.options.lineNumber,
+        readOnly: this.readonly? "nocursor" : false,
+        styleSelectedText: true,
+        line: true,
+        foldGutter: true,
+        gutters: ["CodeMirror-linenumbers", "CodeMirror-foldgutter"],
+        highlightSelectionMatches: { showToken: /\w/, annotateScrollbar: true },
+        mode: 'text/javascript',
+        // hint.js options
+        hintOptions:{
+          // Automatically complete when there is only one match
+          completeSingle: true
+        },
+        // Shortcuts Available in three modes sublime、emacs、vim
+        keyMap: "sublime",
+        matchBrackets: true,
+        showCursorWhenSelecting: true,
+        theme: "default",
+        extraKeys: { "Ctrl": "autocomplete" }
+      }
+    },
+    codemirror() {
+      return this.$refs.cm.codemirror;
+    }
+  },
+  methods: {
+    onCmReady(cm) {
+      console.log('the editor is readied!', cm)
```

Comment:
We should avoid console.log statements in production code. Let's remove these methods unless they're going to be used for something.